ui/app/controllers/application.js

```diff
@@ -0,0 +1,35 @@
+import Ember from 'ember';
+
+const { Controller, computed } = Ember;
+
+export default Controller.extend({
+  error: null,
+
+  errorStr: computed('error', function() {
+    return this.get('error').toString();
+  }),
+
+  errorCodes: computed('error', function() {
+    const error = this.get('error');
+    const codes = [error.code];
+
+    if (error.errors) {
+      error.errors.forEach(err => {
+        codes.push(err.status);
+      });
+    }
+
+    return codes
+      .compact()
+      .uniq()
+      .map(code => '' + code);
+  }),
+
+  is404: computed('errorCodes.[]', function() {
```

Comment:
are you ever going to get 403s because of ACL restrictions? any other error codes to worry about?

Reply:
403s could be a possibility. I think for the most part responses will either be partial or empty though. 

There will also be 501s for features that are in pro/premium SKUs, but I don't think that needs an error page. Just conditional logic to hide/show things.

I could very well be making a 403 error in, like, a week, but for now I'm going with YAGNI.